aries_vcx/src/common/proofs/proof_request.rs

```diff
@@ -60,6 +71,33 @@ impl ProofRequestData {
         Ok(self)
     }
 
+    // Support requested_predicates in 2 formats
+    // Referent-less format, deprecated:
+    // [ { name: 'age', p_type: '>=', p_value: 18 } ]
+    // Explicit referent format:
+    // { 'is_adult': { name: 'age', p_type: '>=', p_value: 18 } }
+    // todo: move String parsing to libvcx_core
+    pub fn set_requested_predicates_as_string(mut self, requested_predicates: String) -> VcxResult<Self> {
+        match serde_json::from_str::<HashMap<String, PredicateInfo>>(&requested_predicates) {
+            Ok(predicates) => self.requested_predicates = predicates,
+            Err(_err) => {
+                warn!("Requested predicates are not in referent format. Trying to parse as array of predicates (deprecated).");
+                let requested_predicates: Vec<PredicateInfo> =
+                    ::serde_json::from_str(&requested_predicates).map_err(|err| {
+                        AriesVcxError::from_msg(
+                            AriesVcxErrorKind::InvalidJson,
+                            format!(
+                                "Invalid Requested Predicates: {:?}, err: {:?}",
+                                requested_predicates, err
+                            ),
+                        )
+                    })?;
+                self = self.set_requested_predicates_as_vec(requested_predicates)?;
+            }
+        }
+        Ok(self)
+    }
```

Comment:
Pretty much same things as for the attributes.

Reply:
dtto